### PR TITLE
add homepage and related components, add routerdom

### PR DESCRIPTION
**Pull request by nathanSamosa**
I have added react-router-dom so remember to install it once merged.

### CODE REVIEW

**Review by molloyconor511 — APPROVED** [review]

File: src/client/App.jsx [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
+    })
+
     return (
-        <div className="App">
-            <h1>Hello, world!</h1>
+        <div className="app">
+
+            <Header isLoggedIn={isLoggedIn} />
+
+            {!isLoggedIn && <Welcomepage />}
```
Comment (polarbear23) [inline src/client/App.jsx:27]: nice conditionals

**Review by bravint — APPROVED** [review]
we need to discuss what we do re. white spaces in the returned jsx, but that aside looks good.

File: src/client/components/CompetitionCard.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import './styling/competition-card.css'
+
+function CompetitionCard() {
```
Comment (polarbear23) [inline src/client/components/CompetitionCard.jsx:6]: Why not arrow functions?

Comment (bravint) [reply src/client/components/CompetitionCard.jsx:6]: why not indeed

File: src/client/components/CompetitionCard.jsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    //const { competition } = props
+
+    //find competitors with the 3 highest scores
+    //(competitionId > many competitors > many participants > many placements > calculate and sum positionMapping)
+
+    let navigate = useNavigate();
+
+    const dummyData = [
+        {
+            name: "Steve",
```
Comment (polarbear23) [inline src/client/components/CompetitionCard.jsx:17]: I like that everywhere I look I find Steves name somewhere beautiful

File: src/client/components/CompetitionCard.jsx (hunk 2 above)
Comment (nathanSamosa) [reply src/client/components/CompetitionCard.jsx:6]: App wasn't an arrow function, so I just kept the same. I do prefer arrow functions though, so happy to change them all for future components.

Comment (polarbear23) [reply src/client/components/CompetitionCard.jsx:6]: I approve looking forward to seeing more arrow functions they are cooler

**Review by polarbear23 — APPROVED** [review]

**Review by vherus — APPROVED** [review]
Looks like some good foundations to build on :)

**Pull request merged**